### DISCUSSION TITLE
some questions to simulate bacnet device.

### DISCUSSION THREAD

**Discussion by jyjker** [bacnet-stack/bacnet-stack#669] [Q&A]
well,I want to use the source code to simulate a bacnet device.  I see many demos in directory called 'apps',use  the struct My_Object_Table[].
![企业微信截图_17177272175321](https://github.com/bacnet-stack/bacnet-stack/assets/87216740/e7a883cb-abd6-4196-8955-72a388a0aeb5)

But this is a statically created structure,if I want to create it dynamically while the program is running,for example, i want to create a new Analog_input object ,there doesn't seem to be a good way to do that in the current code,am I missing something, or is it really only possible to create statically?

**Comment by jyjker** [comment]
i have seen a similar question in anather discussition

**Comment by skarg** [comment]
The apps/server (bin/bacserv) creates an object table for the BACnet object types supported, and many of the objects can be created or deleted dynamically using CreateObject and DeleteObject services which can be used externally. There are sample applications for client apps/create-object (bin/bacco) and apps/delete-object (bin/bacdo).

**Reply by jyjker** [reply]
Yeah,thanks for your patient response! I don't know if that's right,in my opinion,apps/create-object(bin/bacco) and apps/delete-object(bin/bacdo) are applications for sending the commands to operate other devices.It's a little different from what I wanted.Now I want to simulate a bacnet device using my own application,which can dynamically create or delete objects,and modify any properties I want to modify. 
      Using CreateObject and DeleteObeject helps a lot. In the version bacnet-stack-1.3.6 ，many previously missing functions were built,thank you for your effort!
      Best regards
